panes/add-person-to-page.vue

```diff
@@ -0,0 +1,155 @@
+<style lang="sass">
+  @import '../styleguide/inputs';
+  @import '../styleguide/colors';
+  @import '../styleguide/buttons';
+  @import '../styleguide/typography';
+
+  // add-person-to-page is very similar to directory, but with fewer features
+  .add-person-to-page-pane {
+    height: calc(100% - 51px);
+
+    &-input {
+      padding: 4px;
+      &-field {
+        @include input();
+
+        padding: 10px 14px;
+      }
+    }
+
+    &-readout {
+      overflow-y: scroll;
+      overflow-x: hidden;
+      // height is pane height minus search box and bottom button area
+      height: calc(100% - 134px);
+      padding: 0 18px 18px;
+
+      &-list {
+        list-style: none;
+        margin: 0;
+        padding: 0;
+      }
+    }
+
+    &-item {
+      align-items: center;
+      border-bottom: 1px solid $pane-list-divider;
+      cursor: pointer;
+      display: flex;
+      min-height: 53px;
+
+      .user-image {
+        flex: 0 0 32px;
+        height: 32px;
+        margin-right: 15px;
+        width: 32px;
+      }
+
+      .user-name,
+      .user-name-empty {
+        @include primary-text();
+
+        border: none;
+        flex: 1 0 auto;
+        line-height: 1.4;
+        padding-right: 0 15px;
+        // no left padding (it's handled by the image),
+        // so name displays to the left if image doesn't exist
+        text-align: left;
+      }
+
+      .user-name-empty {
+        @include tertiary-text();
+      }
+    }
+  }
+</style>
+
+<template>
+  <div class="add-person-to-page-pane">
+    <div class="add-person-to-page-pane-input">
+      <input
+        type="text"
+        class="add-person-to-page-pane-input-field"
+        placeholder="Search for someone"
+        ref="search"
+        v-model="query"
+        @keyup="onSearchKeyup">
+    </div>
+    <div class="add-person-to-page-pane-readout">
+      <ul class="add-person-to-page-pane-readout-list" ref="list">
+        <li v-for="(user, index) in users" class="add-person-to-page-pane-item" @click.stop.prevent="addPersonToPage(user)">
+          <img v-if="user.imageUrl" class="user-image" :src="user.imageUrl" />
+          <span v-if="user.name" class="user-name">{{ user.name }}</span>
+          <span v-else class="user-name-empty">{{ user.username }}</span>
+        </li>
+      </ul>
+    </div>
+  </div>
+</template>
+
+<script>
+  import _ from 'lodash';
+  import { postJSON } from '../lib/core-data/api';
+  import { searchRoute } from '../lib/utils/references';
+
+  function buildUserQuery(query) {
```

Comment:
nit:
buildUserQuery isn't a UI thing—by pulling it out of the Vue file it would be easier to test

Reply:
Eh, the logic for it is super simple. There's not much to test 😅